integration-tests/mongodb-panache/src/main/resources/application.properties

```diff
@@ -8,3 +8,7 @@ quarkus.mongodb.cl2.write-concern.journal=false
 
 #quarkus.log.category."io.quarkus.mongodb.panache.runtime".level=DEBUG
 quarkus.mongodb.metrics.enabled=true
+
+# avoid CustomMongoDatabaseResolver initialization to used profiles different than MongoDatabaseResolverProfile
+%test.quarkus.arc.exclude-types=io.quarkus.it.mongodb.panache.multitenancy.CustomMongoDatabaseResolver
+%prod.quarkus.arc.exclude-types=io.quarkus.it.mongodb.panache.multitenancy.CustomMongoDatabaseResolver
```

Comment:
It's not very easy to understand why it's needed, maybe a better way would be to move the test to the deployment module where you can control the beans that are inside a test by using `@QuarkusUnitTest`.